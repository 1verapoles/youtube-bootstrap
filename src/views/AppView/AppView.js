export default class AppView {
  constructor(clips) {
    this.clips = clips;
  }

  render() {
    const content = document.createElement('ul');
    content.innerHTML = this.clips.map(title => `<li>${title}</li>`).join('');

    document.body.appendChild(content);
  }
}
