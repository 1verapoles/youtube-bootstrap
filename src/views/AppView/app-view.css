ul {
  list-style: none;
}

ul li {
  display: block;
  margin: 10px;
  background-color: rgb(177, 255, 249);
  border-radius: 10px;
}
