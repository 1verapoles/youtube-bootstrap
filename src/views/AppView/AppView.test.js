import AppView from './AppView';

describe('AppView.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should render correctly', () => {
    const context = {
      clips: [
        'Video about JS',
        'Another video about JS',
        'Undefined video',
      ],
    };

    AppView.prototype.render.call(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
