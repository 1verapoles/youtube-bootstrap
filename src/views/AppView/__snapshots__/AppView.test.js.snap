// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.render Should render correctly 1`] = `"<ul><li>Video about JS</li><li>Another video about JS</li><li>Undefined video</li></ul>"`;
